src/diffusers/models/model_loading_utils.py

```diff
@@ -327,14 +340,19 @@ def _merge_sharded_checkpoints(sharded_ckpt_cached_folder, sharded_metadata):
     # Load tensors from each unique file
     for file_name in files_to_load:
         part_file_path = os.path.join(sharded_ckpt_cached_folder, file_name)
-        if not os.path.exists(part_file_path):
+        if not os.path.exists(part_file_path) and (dduf_entries and part_file_path not in dduf_entries):
```

Comment:
```suggestion
        if not os.path.exists(part_file_path) or (dduf_entries and part_file_path not in dduf_entries):
```
this seem to be `or` to me (because not it only throw an error for dduf) but let me know if not